src/components/Sidebar/BranchTransition.tsx

```diff
@@ -27,12 +80,38 @@ const BranchTransition: FC<BranchTransitionProps> = ({ pathway, transition, upda
         updatePathway={updatePathway}
         isTransition={true}
       />
+      {!useCriteriaSelected && (
+        <SidebarButton
+          buttonName="Use Criteria"
+          buttonIcon={<FontAwesomeIcon icon={faPlus} />}
+          buttonText="Add previously built or imported criteria logic to branch node."
+          onClick={handleUseCriteria}
+        />
+      )}
 
-      <SidebarButton
-        buttonName="Use Criteria"
-        buttonIcon={<FontAwesomeIcon icon={faPlus} />}
-        buttonText="Add previously built or imported criteria logic to branch node."
-      />
+      {useCriteriaSelected && (
+        <>
+          <DropDown
+            id="Criteria"
+            label="Criteria"
+            options={criteriaOptions}
+            onChange={selectCriteriaSource}
+            value={transition.condition?.cql || undefined}
+          />
+          <FormControl variant="outlined" fullWidth>
+            <TextField
+              label="Criteria display"
+              value={transition.condition?.description || ''}
+              variant="outlined"
+              onChange={setCriteriaDisplay}
+              error={
+                transition.condition?.description === undefined ||
+                transition.condition?.description === ''
+              }
```

Comment:
The empty string actually evaluates to false so you should be able to make this
```suggestion
              error={!transition.condition?.description}
```